### PR TITLE
Add ADR Template

### PR DESCRIPTION
**Pull request by benjdlambert**
## Hey, I just made a Pull Request!
- Look like we were missing this :shrug:
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by andrewthauer — APPROVED** [review]
👍

File: docs/architecture-decisions/0000-template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# ADR 0000: [title]
+
+<!-- These documents have names that are short noun phrases. For example, "ADR 1: Deployment on Ruby on Rails 3.0.10" or "ADR 9: LDAP for Multitenant Integration" -->
```
Comment (freben) [inline docs/architecture-decisions/0000-template.md:3]: Update the above two parts to reflect that we call them eg ADR001 with three digits

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
Oh one more thing. These are supposed to have yaml frontmatter right? That should probably be in the template, along with a comment on how to add the file to the index as well

**Comment by benjdlambert** [comment]
> Oh one more thing. These are supposed to have yaml frontmatter right? That should probably be in the template, along with a comment on how to add the file to the index as well

Not sure what you mean here? This file is added because our documentation on adding an ADR [here](https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/index.md#creating-an-adr) references a file that doesn't exist for a template.

**Comment by freben** [comment]
I mean the yaml at the top of files, like the first few lines here https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md

**Comment by benjdlambert** [comment]
> I mean the yaml at the top of files, like the first few lines here https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md

Ah! Now I understand, the preamble for mkdocs. Yep, i'll add it

**Comment by benjdlambert** [comment]
@freben can I get 'hell yeah' to merge?

**Review by freben — APPROVED** [review]
Go go :)

**Review by Rugvip — APPROVED** [review]

**Pull request merged**